File: backend/src/ShriNrityalaya.Domain/Entities/SkillAssessment.cs
using ShriNrityalaya.Domain.Common;

namespace ShriNrityalaya.Domain.Entities;

public class SkillAssessment : AuditableEntity
{
    public Guid StudentId { get; set; }
    public Student Student { get; set; } = null!;

    public Guid TeacherId { get; set; }
    public Guid BatchId { get; set; }
    
    public DateTime AssessmentDate { get; set; }
    public string SkillType { get; set; } = "General"; // Adavu, Margam, Theory
    public string SkillName { get; set; } = string.Empty; // Tatta Adavu, Alarippu
    public string SkillLevel { get; set; } = string.Empty; // Beginner, Intermediate, Advanced
    public string Remarks { get; set; } = string.Empty;
    public int Score { get; set; } // Out of 100
}

File: frontend/lib/features/progress/domain/models/skill_assessment.dart
class SkillAssessment {
  final String? id;
  final String studentId;
  final String teacherId;
  final String batchId;
  final DateTime assessmentDate;
  final String skillType;
  final String skillName;
  final String skillLevel;
  final String remarks;
  final int score;

  SkillAssessment({
    this.id,
    required this.studentId,
    required this.teacherId,
    required this.batchId,
    required this.assessmentDate,
    required this.skillType,
    required this.skillName,
    required this.skillLevel,
    required this.remarks,
    required this.score,
  });

  factory SkillAssessment.fromJson(Map<String, dynamic> json) {
    return SkillAssessment(
      id: json['id'],
      studentId: json['studentId'] ?? '',
      teacherId: json['teacherId'] ?? '',
      batchId: json['batchId'] ?? '',
      assessmentDate: json['assessmentDate'] != null 
          ? DateTime.parse(json['assessmentDate']) 
          : DateTime.now(),
      skillType: json['skillType'] ?? 'General',
      skillName: json['skillName'] ?? '',
      skillLevel: json['skillLevel'] ?? '',
      remarks: json['remarks'] ?? '',
      score: json['score'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'studentId': studentId,
      'teacherId': teacherId,
      'batchId': batchId,
      'assessmentDate': assessmentDate.toIso8601String(),
      'skillType': skillType,
      'skillName': skillName,
      'skillLevel': skillLevel,
      'remarks': remarks,
      'score': score,
    };
  }
}
